embulk-core/src/main/java/org/embulk/spi/time/TimestampFormatter.java

```diff
@@ -139,30 +95,35 @@ public static TimestampFormatter of(final Task task, final Optional<? extends Ti
         public Optional<String> getFormat();
     }
 
-    // Using Joda-Time is deprecated, but the getter returns org.joda.time.DateTimeZone for plugin compatibility.
-    // It won't be removed very soon at least until Embulk v0.10.
-    @Deprecated
+    @Deprecated  // This method will be removed sooner when Joda-Time is removed from Embulk during v0.10.
     public org.joda.time.DateTimeZone getTimeZone() {
-        if (this.delegate == null) {
-            throw new RuntimeException("FATAL: Unexpected execution path of TimestampFormatter without delegate.");
-        }
-        return this.delegate.getTimeZone();
+        return TimeZoneIds.parseJodaDateTimeZone(this.zoneIdString);
     }
 
+    @SuppressWarnings("deprecation")  // https://github.com/embulk/embulk/issues/1292
```

Comment:
looks like suppression like that will have no effect because class `TimestampFormatter` is also market as deprecated.

Reply:
Actualy `@SuppressWarnings("deprecation")` warned. Hmm, there may be some difference in building environment? Let me have another look later.